tests/test_array4.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+import numpy as np
+import amrex
+
+def test_array4_empty():
+    empty = amrex.Array4_double()
+
+    # Check properties
+    assert(empty.size == 0)
+    assert(empty.nComp == 0)
+
+    # assign empty
+    emptyc = amrex.Array4_double(empty)
+    # Check properties
+    assert(emptyc.size == 0)
+    assert(emptyc.nComp == 0)
+
+def test_array4():
+    # from numpy (also a non-owning view)
+    x = np.ones((2, 3, 4,))
+    arr = amrex.Array4_double(x)
+
+    x[1, 1, 1] = 42
+    # TypeError: 'amrex.amrex_pybind.Array4_double' object is not subscriptable
+    # assert(arr[1, 1, 1] == 42)
+
+    # copy to numpy
+    c_arr2np = np.array(arr, copy=True)
```

Comment:
Dumb question -- but I need to learn eventually -- is this the "official" way to define a amrex -> python deepcopy (i.e. using the `np.array` constructor)? Or do we need to define a dedicated deepcopy also?

Reply:
Followup question: do we need to define a non-aliasing constructor (i.e. python -> amrex deepcopy)?

I can imaging that for data analytics applications we might want to keep a time series in python (looking at you machine learning folks) -- for that we need the former (amrex -> python) which is covered by this.

For experimental science applications we frequently have python provide a data source. A python -> amrex deepcopy (probably a simple copy constructor?) would be helpful in ensuring that the data is _owned_ by amrex. This application is far more niche, but not thinking about it now might bite us in the arse later.

Reply:
First question: I added some details above. This is just a view at the moment, the same way as the C++ Array4 is a plain view to non-owned data.